Source/Manager/User/User.swift

```diff
@@ -292,6 +292,38 @@ public class User: UserProtocol {
         isPersistent = true
     }
 
+    private static let keyPrefix = "old-sdk-app-transfer-"
+
+    func storeOnDevice(key: String) throws {
+        do {
+            let tokens = try UserTokensStorage(accessGroup: clientConfiguration.accessGroup).loadTokens()
+
+            if let encoded = try? JSONEncoder().encode(tokens) {
+                UserDefaults.standard.set(encoded, forKey: User.keyPrefix + key)
```

Comment:
`UserDefaults` seems like a poor choice for this. https://cocoacasts.com/ud-11-why-do-you-lose-data-stored-in-user-defaults

At minimum you'd need to flush immediately after setting this.

Reply:
Interesting 🤓

Reply:
Although article says: 

> "Data loss isn't something you need to worry about if you're working with the defaults system. The UserDefaults class ensures any changes are written to disk at the appropriate time without impacting the performance of your application. It goes without saying that unexpected terminations of your application break this behavior, but that applies to any persistence solution."

Reply:
i'll add an invocation to [synchronise](https://developer.apple.com/documentation/foundation/userdefaults/1414005-synchronize)

Reply:
Yes, synchronise